src/database/schema/swaps_v2.ts

```diff
@@ -0,0 +1,56 @@
+import { relations } from 'drizzle-orm';
+import { pgTable, serial, varchar, integer, timestamp, boolean, unique, jsonb, numeric } from 'drizzle-orm/pg-core';
+
+import { chains } from './chains';
+import { tokens } from './tokens';
+
+// Define the swaps table without foreign key constraints on poolId
+export const swapsTableV2 = pgTable(
+  'swaps_v2',
+  {
+    id: serial('id').primaryKey(),
+    chainId: integer('chain_id')
+      .notNull()
+      .references(() => chains.id, { onDelete: 'cascade' }),
+    transactionHash: varchar('transaction_hash', { length: 256 }).notNull(),
+    user: varchar('user', { length: 256 }).notNull(),
+    baseId: varchar('base_id', { length: 256 }).notNull(),
+    quoteId: varchar('quote_id', { length: 256 }).notNull(),
+    poolId: integer('pool_id') // Removed foreign key constraint
+      .notNull(), // You can still keep it as not null if you need to ensure a value is always provided
+    dexType: varchar('dex_type', { length: 256 }).notNull(),
+    poolIdx: varchar('pool_idx', { length: 256 }).notNull(),
+    block: integer('block').notNull(),
+    isBuy: boolean('is_buy').notNull(),
+    amountIn: numeric('amount_in', { precision: 38, scale: 18 }).default('0'),
+    amountOut: numeric('amount_out', { precision: 38, scale: 18 }).default('0'),
```

Comment:
Lets have amount columns as strings because it allows us any amount of decimals and precision.

For a column name - I think it should be same a column which references tokens, so either this should be changed to base_amount / quote_amount or base_id -> token_in_id / quote_id -> token_out_id

```suggestion
    baseAmount: varchar('base_amount', { length: 256 }).notNull().default('0'),
    quoteAmount: varchar('quote_amount', { length: 256 }).notNull().default('0'),
```